### PR TITLE
[DML EP] Add subgraph fusion support

### PR DESCRIPTION
**Pull request by PatriceVignola**

### CODE REVIEW

File: onnxruntime/core/providers/dml/DmlExecutionProvider/src/DmlGraphFusionTransformer.h [hunk 1]
```diff
@@ -2,32 +2,41 @@
...
+    onnxruntime::common::Status ApplyImpl(onnxruntime::Graph& graph,
+                                            bool& modified,
+                                            int graph_level,
+                                            const onnxruntime::logging::Logger& logger) const final;
 
-	public:
-		inline const static char* const DML_GRAPH_FUSION_NODE_NAME_PREFIX = "DmlFusedNode_";
-		inline const static char* const DML_GRAPH_FUSION_NODE_DOMAIN = "DmlFusedNodeDomain";
+    onnxruntime::common::Status DmlGraphFusionTransformer::ApplyImplHelper(
+        onnxruntime::Graph& graph,
```
Comment (sumitsays) [inline onnxruntime/core/providers/dml/DmlExecutionProvider/src/DmlGraphFusionTransformer.h:33]: [nit] No need to specify the class name in the declaration as the method is declared within the class, unless I am missing something.

Comment (PatriceVignola) [reply onnxruntime/core/providers/dml/DmlExecutionProvider/src/DmlGraphFusionTransformer.h:33]: Not missing anything, just bad copy paste from the C++ file :)

**Review by sumitsays — APPROVED** [review]
:shipit:

**Review by fdwr — COMMENTED** [review]
👍

**Pull request merged**